confluent_platform/datadog_checks/confluent_platform/data/metrics.yaml

```diff
@@ -0,0 +1,1276 @@
+        - kafka.server:type=ReplicaManager,name=UnderReplicatedPartitions
+      attribute:
+        Value:
+          alias: $domain.$type.$name
+          metric_type: gauge
+
+  # Kafka Broker Controller Metrics (Yammer Metric Gauge)
+  - include:
+      domain: kafka.controller
+      bean:
+        # Number of partitions that don’t have an active leader and are hence not writable or readable. Alert if value is greater than 0.
+        - kafka.controller:type=KafkaController,name=OfflinePartitionsCount
+        # Number of active controllers in the cluster. Alert if the aggregated sum across all brokers in the cluster is anything other than 1 because there should be exactly one controller per cluster.
+        - kafka.controller:type=KafkaController,name=ActiveControllerCount
+      attribute:
+        Value:
+          alias: $domain.$name
+          metric_type: gauge
+
+  # Kafka Broker Server Metrics Regex (Yammer Metric Gauge)
+  - include:
+      domain: kafka.server
+      bean_regex:
+        # Lag in number of messages per follower replica. This is useful to know if the replica is slow or has stopped replicating from the leader.
+        - kafka\.server:type=FetcherLagMetrics,name=ConsumerLag,.*
+      attribute:
+        Value:
+          alias: $domain.fetcher_lag.$name
+          metric_type: gauge
+
+  # Kafka Broker Cluster Metrics Regex (Yammer Metric Gauge)
+  - include:
+      domain: kafka.cluster
+      bean_regex:
+        # Number of partitions whose in-sync replicas count is less than minIsr. These partitions will be unavailable to producers who use acks=all.
+        - kafka\.cluster:type=Partition,name=UnderMinIsr,.*
+      attribute:
+        Value:
+          alias: $domain.$type.$name
+          metric_type: gauge
+
+  # Kafka Broker Topic Metrics (Yammer Metric Meter)
+  - include:
+      domain: kafka.server
+      bean:
+        # Aggregate incoming byte rate.
+        - kafka.server:type=BrokerTopicMetrics,name=BytesInPerSec
+        # Aggregate outgoing byte rate.
+        - kafka.server:type=BrokerTopicMetrics,name=BytesOutPerSec
+        # Aggregate rejected byte rate.
+        - kafka.server:type=BrokerTopicMetrics,name=BytesRejectedPerSec
+        # Fetch request rate for requests that failed.
+        - kafka.server:type=BrokerTopicMetrics,name=FailedFetchRequestsPerSec
+        # Produce request rate for requests that failed.
+        - kafka.server:type=BrokerTopicMetrics,name=FailedProduceRequestsPerSec
+        # Aggregate incoming message rate.
+        - kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec
+        # Fetch request rate.
+        - kafka.server:type=BrokerTopicMetrics,name=TotalFetchRequestsPerSec
+        # Produce request rate.
+        - kafka.server:type=BrokerTopicMetrics,name=TotalProduceRequestsPerSec
+      attribute:
+        Count:
+          alias: $domain.topic.$name.rate
+          metric_type: rate
```

Comment:
Can't find docs about the `rate` JMX metric type. Based on the names and description of those metrics I would have expected those to be gauges (no need to compute any time differences).

Reply:
Yammer Metric `Meter` have multiples attributes: https://github.com/infusionsoft/yammer-metrics/blob/master/metrics-core/src/main/java/com/codahale/metrics/Meter.java#L80-L82

The value of count attribute is monotonically increasing.

This is likely the related code about how rates are processed before being sent to `dogstatsd`:
https://github.com/AlexandreYang/jmxfetch/blob/13be9d19a9a497032199b2cd594e9e6ca090abe8/src/main/java/org/datadog/jmxfetch/reporter/Reporter.java#L121-L150

I'm following the "convention" used in Kafka https://github.com/DataDog/integrations-core/blob/6d0c1986c04e7d5b3a3ee4e9ad1c50623de65ad2/kafka/datadog_checks/kafka/data/metrics.yaml#L328-L334 where you can meet several occurrences of `metric_type: rate`

But it's probably better if there is a proper type on `jmxfetch` side, instead of defaulting to this rate/counter aggregator.

The backend type is `gauge`.

Reply:
Ok got it, so that attribute is monotonically increasing thus the rate 👍